cdc/redo/manager.go

```diff
@@ -567,16 +600,55 @@ func (m *ManagerImpl) bgUpdateLog(ctx context.Context, errCh chan<- error) {
 				for _, row := range cache.rows {
 					logs = append(logs, RowToRedo(row))
 				}
-				err := m.writer.WriteLog(ctx, cache.tableID, logs)
-				if err != nil {
-					handleErr(err)
-				}
+				err = m.writer.WriteLog(ctx, cache.tableID, logs)
 				m.metricWriteLogDuration.Observe(time.Since(start).Seconds())
 			case model.MessageTypeResolved:
 				m.onResolvedTsMsg(cache.tableID, cache.resolvedTs)
 			default:
 				log.Debug("handle unknown event type")
 			}
 		}
+		if err != nil {
+			log.Warn("redo manager writer meets write or flush fail",
+				zap.String("namespace", m.changeFeedID.Namespace),
+				zap.String("changefeed", m.changeFeedID.ID),
+				zap.Error(err))
+			break
+		}
+	}
+
+	// NOTE: the goroutine should never exit until the err is put into errCh successfully
+	// or the context is canceled.
+	select {
+	case <-ctx.Done():
+	case errCh <- err:
+	}
```

Comment:
I think this is a deadlock risk.

Reply:
Owner and processor will cancel `ctx` as expected so here the `select` can finish finally.